linux/usage/systemd.md

```diff
@@ -1,57 +1,101 @@
 # systemd
 
-In order to have a command or program run when the Pi boots, you can add it as a service. Once this is done, you can start/stop enable/disable from the linux prompt.
+Raspberry Pi OS uses [`systemd`](https://www.freedesktop.org/wiki/Software/systemd/) to manage the running of services, including controlling what starts when Linux boots.
 
-## Creating a service
+## `rc.local` under `systemd`
+Unlike on older versions of Linux, `systemd` starts services based on their dependencies: services are no longer started in a predefined order. This means it is no longer guaranteed that `rc.local` will run once all other services have started. You should therefore not use `rc.local` to start programs at boot.
 
-On your Pi, create a .service file for your service, for example:
+## Service definitions
+`Systemd` defines each service in a separate file. Under `systemd` services are a type of 'unit'. Units are system resources under the control of `systemd`. The basic format of a service definition is as follows:
 
-myscript.service
+```
+[Unit]
+Description=...
+...
+
+[Service]
+...
+
+[Install]
+...
+```
+
+User services should be placed in `/etc/systemd/user`: the filename dictates the service name. For example, a file named `monitoring.service` would define the service named `monitoring`.
+
+### Example - one-time service
+One-time services are used when you wish to run a program which does some work, then exits: this is in contrast to a service which stays running all the time. Consider the following service definition:
+
+```
+[Unit]
+Description=Run my program once at boot time
+After=local-fs.target network-online.target
```

Comment:
Generally thumps up for `network-online.target` over `network.target`, which has the meaning that is usually wanted.

But note that `network-online.target` is an active target, hence it needs to be pulled in by systemd units that want to order themselves against it. Change to:
```
Wants=network-online.target
After=local-fs.target network-online.target
```
Source: https://www.freedesktop.org/software/systemd/man/systemd.special.html#network-online.target

Reply:
Thanks. I've added Wants=network-online.target, and removed local-fs.target from the After line to simplify things slightly.

Reply:
Does `network-online.target` work for Pis without an Ethernet or Wifi network connection, e.g. the Model A+ or Pi Zero?

> ... and removed local-fs.target from the After line ...

But the text underneath still mentions `local-fs.target` ?

Reply:
These are only examples. If folk have no networking, it should be obvious they don't need that line.